ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Overview/NodeCvesOverviewPage.tsx

```diff
@@ -55,12 +56,12 @@ import NodesTable, {
 } from './NodesTable';
 import { useNodeCveEntityCounts } from './useNodeCveEntityCounts';
 
-const searchFilterConfig = {
-    Node: nodeSearchFilterConfig,
-    NodeCVE: nodeCVESearchFilterConfig,
-    'Node Component': nodeComponentSearchFilterConfig,
-    Cluster: clusterSearchFilterConfig,
-};
+const searchFilterConfig = createSearchFilterConfig([
+    nodeSearchFilterConfig,
+    nodeCVESearchFilterConfig,
+    nodeComponentSearchFilterConfig,
+    clusterSearchFilterConfig,
+]);
```

Comment:
You're right. Changing it to an object does simplify things and seems like an improvement we can include. The "pickAttrs" function would also be useful and work similarly to the "get[Entity]Attributes" function. The only consideration is that to show just the operating system attribute, I'd now have to pass "OperatingSystem" as the key value. Would we even need do that anymore. If we also export all the individual attributes, maybe we can just build it out here:

```
import { Name, Type } from 'Components/CompoundSearchFilter/attributes/cluster';

const searchFilterConfig = {
    displayName: 'Cluster',
    searchCategory: 'CLUSTERS',
    attributes: { Name, Type },
};
```

Reply:
One issue that pops up from the object approach vs. the array approach is how we look things up. I'm trying out the changes you suggested and looks like any attributes that have a space between them (e.g. "Operating system", "Discovered time", etc.) don't show their input components. This may require a few tweaks to either the logic for rendering or how we structure our config types. But it shouldn't be a big issue though.

Reply:
One thing I'm considering is if we want to make the config object an array:

```
// Before
const config = {
    Cluster: {
        displayName: 'Cluster',
        searchCategory: 'CLUSTERS',
        attributes: { Name, Type },
    }
}
```

```
// After
const config = [
    {
        displayName: 'Cluster',
        searchCategory: 'CLUSTERS',
        attributes: [Name, Type],
    }
]
```

Pros: 
- Simpler
- No need to specify the entity key anymore. The component will internally determine the dropdown options using the displayName.

Cons:
- A bit more work to modify some logic in utils and compound search component files.

Reply:
All good ideas, and even simpler and easier to read than the utility function change I suggested.